proto/osmosis/incentives/group_gauge.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+package osmosis.incentives;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "osmosis/lockup/lock.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v19/x/incentives/types";
+
+// SplittingPolicy determines the way we want to split incentives in groupGauges
+enum SplittingPolicy {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  Volume = 0;
+  Evenly = 1;
+}
+
+// Note that while both InternalGaugeInfo and InternalGaugeRecord could
+// technically be replaced by DistrInfo and DistrRecord from the pool-incentives
+// module, we create separate types here to keep our abstractions clean and
+// readable (pool-incentives distribution abstractions are used in a very
+// specific way that does not directly relate to gauge logic). This also helps
+// us sidestep a refactor to avoid an import cycle.
+message InternalGaugeInfo {
+  string total_weight = 1 [
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.moretags) = "yaml:\"total_weight\"",
+    (gogoproto.nullable) = false
+  ];
+  repeated InternalGaugeRecord gauge_records = 2
+      [ (gogoproto.nullable) = false ];
+}
+
+message InternalGaugeRecord {
+  option (gogoproto.equal) = true;
+
+  uint64 gauge_id = 1 [ (gogoproto.moretags) = "yaml:\"gauge_id\"" ];
+  string current_weight = 2 [
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false
+  ];
+  string cumulative_weight = 3 [
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false
+  ];
```

Comment:
Could you add comments explaining the difference between how we use current and cumulative weights?